src/sqlodbc/opensearch_communication.cpp

```diff
@@ -1046,17 +1075,41 @@ std::string OpenSearchCommunication::GetClusterName() {
 }
 
 /**
- * @brief Sets URL endpoint for SQL plugin. On failure to
- * determine appropriate endpoint, value is set to SQL_ENDPOINT_ERROR_STR
- * 
+ * @brief Sets URL endpoint for the SQL plugin.
+ * Sets it to SQL_ENDPOINT_ERROR if an appropriate
+ * endpoint could not be determined.
  */
 void OpenSearchCommunication::SetSqlEndpoint() {
+
+    // Serverless Elasticsearch is not supported.
+    if (is_serverless) {
+        sql_endpoint = SQL_ENDPOINT_OPENSEARCH;
+        return;
+    }
+
     std::string distribution = GetServerDistribution();
     if (distribution.empty()) {
-        sql_endpoint = SQL_ENDPOINT_ERROR_STR;
-    } else if (distribution.compare("opensearch") == 0) {
-        sql_endpoint = "/_plugins/_sql";
+        sql_endpoint = SQL_ENDPOINT_ERROR;
+    } else if (distribution == "opensearch") {
```

Comment:
defined "opensearch" as a static constant on the top of the file

Reply:
Thanks. Done ✅ 

Aside: Would probably be more robust to also add a `DISTRIBUTION_ELASTICSEARCH` constant as well, and then restructure to below. But I don't know what the constant would be for ElasticSearch, and don't want to get side-tracked by this.

```
std::string distribution = GetServerDistribution();
  if (distribution == DISTRIBUTION_OPENSEARCH) {
    sql_endpoint = SQL_ENDPOINT_OPENSEARCH;
  } else if (distribution == DISTRIBUTION_ELASTICSEARCH) {
    sql_endpoint = SQL_ENDPOINT_ELASTICSEARCH;
  } else {
    sql_endpoint = SQL_ENDPOINT_ERROR;
  }
```